File: always-test-ads/app/src/main/java/com/codedogg/alwaystestads/MainActivity.java
package com.codedogg.alwaystestads;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // load admob ad
        AdView admobBanner = findViewById(R.id.ad_view);
        admobBanner.loadAd(new AdRequest.Builder().build());
    }
}

File: dialogfragment-with-viewmodel/app/src/main/java/com/codedogg/smartdialog/MainActivity.kt
package com.codedogg.smartdialog

import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.codedogg.smartdialog.purchasedialog.view.PremiumFeaturesFragment

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val purchaseButton = findViewById<Button>(R.id.purchase)
        purchaseButton.setOnClickListener {
            PremiumFeaturesFragment().show(supportFragmentManager, null)
        }
    }
}